ZENKO-2833-testing-list-buckets-components-and-actions

romainlancia opened the pull request.

romainlancia committed:
ZENKO-2833 feat: testing list bucket actions and components
[+] new file test for Bucket component
[+] new file test for BucketList component
[+] new tests for for listBuckets and listBucketSuccess actions in s3bucket.test file

bert-e commented:
# Hello lebroz,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @nicolas2bert.

nicolas2bert commented on src/react/actions/__tests__/s3bucket.test.jsx:
```diff
@@ -43,6 +53,16 @@ describe('s3bucket actions', () => {
             ],
         },
+        {
+            it: 'listBuckets: should return list of buckets',
```
"Should handle error" test?

romainlancia replied:
done

nicolas2bert commented on src/react/databrowser/buckets/__tests__/BucketList.test.jsx:
```diff
@@ -0,0 +1,82 @@
+                expect(bucketRow.find('tr').parent().prop('isSelected')).toBe(true);
+            }
+        });
+
```
NIT: extra line.

romainlancia replied:
done

nicolas2bert commented on src/react/actions/__tests__/s3bucket.test.jsx:
```diff
@@ -2,13 +2,23 @@ import * as actions from '../s3bucket';
 describe('s3bucket actions', () => {
+    const syncTests = [
+        {
+            it: 'should return OPEN_ACCOUNT_DELETE_DIALOG action',
```
Incorrect description.

romainlancia replied:
done

nicolas2bert commented on src/react/databrowser/buckets/__tests__/BucketList.test.jsx:
```diff
@@ -0,0 +1,82 @@
+import { reduxMount } from '../../../utils/test';
+
+describe('BucketList', () => {
+    const buckets = [{
```
buckets should be an immutable List.

romainlancia replied:
done

nicolas2bert commented on src/react/databrowser/buckets/__tests__/BucketList.test.jsx:
```diff
@@ -0,0 +1,82 @@
+            </MemoryRouter>,
+        );
+
+        const bucketRows = component.find('Row');
```
If you import and find the `Row` components directly, you won't have to  `.find('tr').parent()`.

romainlancia replied:
done

nicolas2bert commented on src/react/databrowser/buckets/__tests__/BucketList.test.jsx:
```diff
@@ -0,0 +1,82 @@
+
+        const bucketRows = component.find('Row');
+        bucketRows.forEach((bucketRow, index) => {
+            if (buckets[index].Name === 'bucket1') {
```
We don't know that `bucketRows` and `buckets` arrays will stay sorted in the same way. 
If we decide to sort our `<Row>`s by bucket name for example, code will still be correct (meaning that the right `<Row>` will be selected) but the tests will fail.

romainlancia replied:
done

nicolas2bert commented on src/react/databrowser/buckets/__tests__/Buckets.test.jsx:
```diff
@@ -0,0 +1,58 @@
+        Name: 'bucket2',
+    }];
+
+    it('should display EmptyStateContainer if no buckets is present', () => {
```
NIT: no bucket

romainlancia replied:
done

nicolas2bert commented on src/react/databrowser/buckets/__tests__/BucketList.test.jsx:
```diff
@@ -0,0 +1,82 @@
+    it('should select row if the bucket name specified in the parameter matches one of the bucket names listed', () => {
+        const { component } = reduxMount(
+            <MemoryRouter>
+                <BucketList buckets={buckets} locations="" selectedBucketName="bucket2"/>
```
`locations` can not be of type string. It is an "object as map". Its zero-value can be set to `{}`

romainlancia replied:
done

nicolas2bert commented on src/react/databrowser/buckets/__tests__/BucketList.test.jsx:
```diff
@@ -0,0 +1,82 @@
+    it('should select no row if there is no bucket name specified in the parameter', () => {
+        const { component } = reduxMount(
+            <MemoryRouter>
+                <BucketList buckets={buckets} locations="" selectedBucketName=""/>
```
Same comment it for `locations`.

romainlancia replied:
done

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @nicolas2bert.




The following reviewers are expecting changes from the author, or must review again:

* @nicolas2bert

nicolas2bert commented:
You need your JIRA ticket number in your commit.

romainlancia commented:
I just reword the commit message with the ticket number inside

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @nicolas2bert.

romainlancia committed:
ZENKO-2869 feat: test delete bucket (#44)

[+] openBucketDialog test
[+] closeBucketDialo test
[+] deletBucket action case success test
[+] deleteBucket action case error test

romainlancia commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue ZENKO-2833.

Goodbye lebroz.
